File: Offer/Java/Offer68.java
//public class Offer68 {
//    class TreeNode {
//        int val;
//        TreeNode left;
//        Offer55.TreeNode right;
//
//        TreeNode() {
//        }
//
//        TreeNode(int val) {
//            this.val = val;
//        }
//
//        TreeNode(int val, TreeNode left, Offer55.TreeNode right) {
//            this.val = val;
//            this.left = left;
//            this.right = right;
//        }
//    }
//
//    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
//        while (root != null){
//            if (root.val < p.val && root.val < q.val){
//                root =  root.right;
//            }
//            else if(root.val > p.val && root.val > q.val){
//                root = root.left;
//            }
//            else {
//                break;
//            }
//        }
//        return root;
//    }
//}

File: Offer/C++/offer68.cc
#include<iostream>
using namespace std;

struct TreeNode{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x): val(x),left(NULL),right(NULL){}
};

class Solution {
public:
    TreeNode* ans(TreeNode* root, TreeNode* p, TreeNode* q){
        if(root->val < p->val){
            return ans(root->right,p,q);
        }
        if(root->val > q->val){
            return ans(root->left,p,q);
        }
        return root;
    }
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        if(p->val > q->val){
            return ans(root,q,p);
        }
        return ans(root,p,q);
    }
};